### PR TITLE
Architecture docs: add arch diagram and lingo updates

### PR DESCRIPTION
**Pull request by ebidel**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -2,20 +2,15 @@
...
-
-![graph of lighthouse-core module dependencies](https://cloud.githubusercontent.com/assets/39191/19367685/04d4336a-9151-11e6-9ebb-3b87bdb09a4c.png)
+## Components & Terminology
 
-`npm install -g js-vd; vd --exclude "node_modules|third_party|fs|path|url|log" lighthouse-core/ > graph.html`
+* **Driver** - Interfaces with [Chrome Debugging Protocol](https://developer.chrome.com/devtools/docs/debugger-protocol)  ([API viewer](https://chromedevtools.github.io/debugger-protocol-viewer/))
+* **Gatherers** - Requests data from the browser or collects information from the page (and maybe post-processing)
+  * **Artifacts** - output of a gatherer
+* **Audits** - Tests for something. Either a pass/fail check or calculates a score.
+* **Categories** - Grouping audit results into a user-facing section of the report (eg. `Best Practices`). Appies weighting and overall scoring to the section.
```
Comment (paulirish) [inline docs/architecture.md]: Applies

File: docs/architecture.md [hunk 2]
```diff
@@ -2,20 +2,15 @@
 
 _Some incomplete notes_
 
-## Components
+![Lighthouse Architecture](https://raw.githubusercontent.com/GoogleChrome/lighthouse/master/assets/architecture.jpg)
 
-* **Driver** - Interfaces with [Chrome Debugging Protocol](https://developer.chrome.com/devtools/docs/debugger-protocol)  ([API viewer](https://chromedevtools.github.io/debugger-protocol-viewer/))
```
Comment (paulirish) [inline docs/architecture.md]: nuke the API viewer link since these two things now redirect to the same place

File: docs/architecture.md [hunk 3]
```diff
@@ -2,20 +2,15 @@
...
-* **Aggregations** - Pulling audit results, grouping into user-facing components (eg. `install_to_homescreen`) and applying weighting and overall scoring.
-
-### Internal module graph
-
-![graph of lighthouse-core module dependencies](https://cloud.githubusercontent.com/assets/39191/19367685/04d4336a-9151-11e6-9ebb-3b87bdb09a4c.png)
+## Components & Terminology
 
-`npm install -g js-vd; vd --exclude "node_modules|third_party|fs|path|url|log" lighthouse-core/ > graph.html`
+* **Driver** - Interfaces with [Chrome Debugging Protocol](https://developer.chrome.com/devtools/docs/debugger-protocol)  ([API viewer](https://chromedevtools.github.io/debugger-protocol-viewer/))
+* **Gatherers** - Requests data from the browser or collects information from the page (and maybe post-processing)
```
Comment (brendankenny) [inline docs/architecture.md]: > Requests data from the browser or collects information from the page

is there another way to phrase this? (collecting information from the page is *also* data requested from the browser). Should it mention that it's collected via the driver?

File: docs/architecture.md [hunk 4]
```diff
@@ -2,20 +2,15 @@
...
+  * **Artifacts** - output of a gatherer
+* **Audits** - Tests for something. Either a pass/fail check or calculates a score.
```
Comment (brendankenny) [inline docs/architecture.md]: "Tests for something using the artifacts as inputs" or something?

File: docs/architecture.md (hunk 3 above)
Comment (paulirish) [inline docs/architecture.md]: > Requests data from the browser, via the driver. Minimal post-processing.

File: docs/architecture.md (hunk 4 above)
Comment (paulirish) [inline docs/architecture.md]: > Using the artifacts as input, the audits do core evaluation and pass/fail/scoring.

let's also add this guy.. (probably as a sub-bullet to audits?):

>  * **Computed Artifacts** - Generated on-demand from artifacts, these add additional meaning, and are often shared amongst multiple audits.

**Comment by paulirish** [comment]
image 👍 👍 👍 👍 👍

File: docs/architecture.md (hunk 1 above)
Comment (ebidel) [inline docs/architecture.md]: done

File: docs/architecture.md (hunk 3 above)
Comment (ebidel) [inline docs/architecture.md]: "Uses Driver to collect information about the page. Minimal post-processing."

File: docs/architecture.md (hunk 4 above)
Comment (ebidel) [inline docs/architecture.md]: "Using the Artifacts as input, Audits evaluate a test and assign pass/fail/scoring."

Done

**Pull request merged**